chrome/content/brief-overlay.js

```diff
@@ -192,12 +194,14 @@ var Brief = {
             Brief.constructTooltip();
     },
 
-    updateStatus: function Brief_updateStatus() {
-        if (!Brief.toolbarbutton)
+    updateStatus: function Brief_updateStatus(toolbarbutton) {
+        if (toolbarbutton == undefined)
+            toolbarbutton = Brief.toolbarbutton.forWindow(window);
```

Comment:
> Thus the updateStatus function inside the global Brief variable has access to the current window object.

Exactly what I'm talking about... seems that the global Bief's `updateStatus` updates the badge in exactly one of the Firefox windows, while the others are left stale. I'll look into this in more details.